lib/super_diff/recursion_guard.rb

```diff
@@ -4,11 +4,25 @@ module SuperDiff
   module RecursionGuard
     RECURSION_GUARD_KEY = "super_diff_recursion_guard_key".freeze
     PLACEHOLDER = "∙∙∙".freeze
+    RECURSION_GUARD_COUNTER_KEY = "super_diff_recursion_guard_counter".freeze
 
     def self.guarding_recursion_of(*objects, &block)
+      # recursion_counter is being added as a crude fix to the problem described in
+      # https://github.com/mcmire/super_diff/issues/160
+      # If that issue receives a more elegant solution, this can be removed from
+      # the Tidelift version of the code.
+      Thread.current[RECURSION_GUARD_COUNTER_KEY] ||= 0
+      Thread.current[RECURSION_GUARD_COUNTER_KEY] = Thread.current[
+        RECURSION_GUARD_COUNTER_KEY
+      ] + 1
```

Comment:
did you want to subtract 1 on line 42? Otherwise I think you count adjacent moves as a +1:

```
SuperDiff.inspect_object({
    first_level_1: { second_level_1: "apples" }, 
    first_level_2: { second_level_2: "oranges"}
  }, 
  as_lines: false
)
```

this is the output I get when debugging the levels (before it runs the given block):

| object      | "super_diff_recursion_guard_key" |
| ----------- | ----------- |
| first_level_1      | 1       |
| {:second_level_1=>"apples"}   | 2        |
| {:second_level_1=>"apples"}   | 3        |
| "apples"   | 4     |
| "apples"   | 5        |
| {:second_level_2=>"oranges"}   | 6        |
| {:second_level_2=>"oranges"}   | 7        |
| "oranges"   | 8        |
| "oranges"  | 9        |

I think we'd want to subtract one for extreme cases like this where it's not recursing: `[1,2,3,4,5]`

Reply:
(or maybe an `ensure` block would work too?)